pkg/build/controller/strategy/util.go

```diff
@@ -193,6 +164,25 @@ func addOriginVersionVar(output *[]kapi.EnvVar) {
 	*output = append(*output, version)
 }
 
+// addOutputEnvVars adds env variables that provide information about the output
+// target for the build
+func addOutputEnvVars(buildOutput *kapi.ObjectReference, output *[]kapi.EnvVar) error {
+	outputVars := []kapi.EnvVar{}
+
+	if buildOutput != nil {
```

Comment:
We can test for the opposite condition to have [less indented code](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow):

```go
// addOutputEnvVars adds to env environment variables that provide information
// about the buildOutput target.
func addOutputEnvVars(buildOutput *kapi.ObjectReference, env *[]kapi.EnvVar) error {
	if buildOutput == nil {
		// output must always be a DockerImage type reference at this point.
		return nil
	}

	ref, err := imageapi.ParseDockerImageReference(buildOutput.Name)
	if err != nil {
		return err
	}
	registry := ref.Registry
	ref.Registry = ""
	image := ref.String()

	outputVars := []kapi.EnvVar{
		{Name: "OUTPUT_REGISTRY", Value: registry},
		{Name: "OUTPUT_IMAGE", Value: image},
	}

	*env = append(*env, outputVars...)
	return nil
}
```